Sources/Apollo/CacheReference.swift

```diff
@@ -0,0 +1,22 @@
+import Foundation
+
+/// A reference to a cache record.
+public struct CacheReference {
```

Comment:
Apollo Android uses `CacheReference` too. I'd be in for renaming to `EntityReference` as we're still at a moment where we can rename stuff on Android. Feels like that would be more adequate. It's a reference to an Entity, not to a Cache. Or maybe `ObjectReference` ? (We don't have entity anywhere in the Android codebase)

Reply:
@designatednerd What's your opinion on the naming of this? Entity, Object, something else?

Reply:
@martinbonnin Is your point that it's a reference to an entity _within_ the cache rather than to the cache itself? I think that's valid, but `Entity` by itself is so overloaded I might go with something like `CacheEntityReference`.

Reply:
> Is your point that it's a reference to an entity within the cache rather than to the cache itself?

Yes it was! Rereading that, I'm certainly overthinking this. `CacheReference` sounds reasonable. 
And now back to overthinking this... What's the difference between a `CacheReference` and a `CacheKey` ? Could we merge both somehow?

Reply:
My recollection is that the key is the key used to access stored data vs the reference is the actual stored data - @AnthonyMDev is that about right?

Reply:
I got curious and tried to merge `CacheKey` and `CacheReference`. Turns out it's working well: https://github.com/apollographql/apollo-android/pull/3157

The only reason I can think of having two different data structures in Kotlin is if we want to make `CacheKey` a value class that is inlined to a `String` for performance reasons. If we do that, we can't check the type of `CacheKey` anymore which means everything becomes a string and we can't serialize the CacheKey to their special String value. 

But besides that I actually like the reduced mental load and API surface.

Reply:
Hmmmmm. I'm going to look into this. Thanks for that @martijnwalraven 

> My recollection is that the key is the key used to access stored data vs the reference is the actual stored data - @AnthonyMDev is that about right?

No, the reference isn't the actual stored data. It's just a wrapper type for the CacheKey as a `String`. It indicates that the raw String data isn't just a regular raw a String property on the object, but is a `String` representing the reference to the cache key of another object.

Reply:
Gooooot it, thank you for clarifying